core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -314,7 +316,7 @@ private[spark] class ExecutorAllocationManager(
       // Do not change our target while we are still initializing,
       // Otherwise the first job may have to ramp up unnecessarily
       0
-    } else if (maxNeeded < numExecutorsTarget) {
+    } else if (maxNeeded <= numExecutorsTarget) {
       // The target number exceeds the number we actually need, so stop adding new
```

Comment:
I'm curious if we needed the = here?  It doesn't appear that it should be needed but perhaps there is corner case.  I don't think it hurts anything either so more curious if there is an existing bug.

If we do keep the = we should update the comment here for is the number exceeds or is equal